airbyte-db/db-lib/build.gradle

```diff
@@ -12,9 +12,9 @@ dependencies {
     implementation project(':airbyte-json-validation')
     implementation project(':airbyte-config:config-models')
     implementation libs.flyway.core
+    implementation libs.platform.testcontainers.postgresql
 
     // Mark as compile only to avoid leaking transitively to connectors
-    compileOnly libs.platform.testcontainers.postgresql
     compileOnly libs.connectors.testcontainers.mysql
```

Comment:
Hey @jdpgrailsdev, I noticed this was changed to `compileOnly` explicitly in https://github.com/airbytehq/airbyte/pull/13642 - I'm not too familiar with java deps and how they may affect the connectors, would love for your thoughts on this

Reply:
@pedroslopez This was done to address another issue caused by letting the testcontainer dependency to be marked as `implementation`.  That scope will cause the dependency to be exposed as a transitive dependency by any other project that depends on this project.  A better approach would be to add a configuration specifically for that Gradle task so we can provide it with the dependencies that it needs to run without exposing those dependencies to other projects that depends on this library.

Reply:
@jdpgrailsdev I thought the keyword that causes transitive dependency is `api` and `implementation` is ok. Refreshed my memory [here](https://discuss.gradle.org/t/is-it-recommended-to-use-compileonly-over-implementation-if-another-module-use-implementation-already/26699/2). 

Does this break something today on the connector side? If not, would you be ok putting back for now? Not having this means our current DB migrations won't work i.e. cannot write a new migration. I'm also open to other ideas to fix this!

Reply:
> @jdpgrailsdev I thought the keyword that causes transitive dependency is `api` and `implementation` is ok. Refreshed my memory [here](https://discuss.gradle.org/t/is-it-recommended-to-use-compileonly-over-implementation-if-another-module-use-implementation-already/26699/2). I might be misunderstanding though.
> 
> Does this break something today on the connector side? If not, would you be ok putting back for now? Not having this means our current DB migrations won't work i.e. cannot write a new migration. I'm also open to other ideas to fix this!

@davinchia You are correct/my mistake.  `implementation` scope should be fine for a java-library module.  I would still recommend a separate configuration for dependencies that are used by a specific task and are not really needed for compilation of the project.  Something like:

```
configurations {
    migrations.extendsFrom implementation
}

dependencies {
    migrations libs.platform.testcontainers.postgresql
    migrations libs.connectors.testcontainers.mysql
    ...
} 

task(dumpConfigsSchema, dependsOn: 'classes', type: JavaExec) {
    main = 'io.airbyte.db.instance.development.MigrationDevCenter'
    classpath = sourceSets.main.migrationsClasspath
    args 'configs', 'dump_schema'
} 
```

That should allow us to remove the compileOnly dependencies and avoid any accidental transitive dependency issue should someone accidentally change the scope.

Reply:
Got it. Thanks for the explanation + example to fix!